src/ol/collection.js

```diff
@@ -225,6 +249,20 @@ ol.Collection.prototype.updateLength_ = function() {
 
 
 /**
+ * @private
+ * @param {T} elem Element.
+ * @param {number=} opt_except Optional index to ignore.
+ */
+ol.Collection.prototype.assertUnique_ = function(elem, opt_except) {
+  for (var i = 0, ii = this.array_.length; i < ii; ++i) {
+    if (this.array_[i] === elem && i !== opt_except) {
+      throw new Error('Duplicate item added to a unique collection');
```

Comment:
Yeah, was thinking the same.  There is a related issue with our version detection that I'd like to look into (links in error messages from our custom builds are wrong - this may no longer be a problem with the ol package).  This issue has plagued us in production for a long time.  At first it was awkward to properly identify due to a useless stack trace and odd failure point.  With the ol package we have a good stack trace but puzzling failure point.  I'm wanting this to fail at the right point and with a clear message to save others the same pain.

I'm compelled by the byte saving argument.  But if we are really committed to that, we'd get rid of ol.Collection entirely.